src/raftstore/store/store.rs

```diff
@@ -294,13 +294,18 @@ impl<T, C> Store<T, C> {
 
     fn clear_stale_meta(&mut self, wb: &mut WriteBatch, region: &metapb::Region) {
         let raft_key = keys::raft_state_key(region.get_id());
-        let handle = rocksdb::get_cf_handle(&self.engine, CF_RAFT).unwrap();
-        if self.engine.get_cf(handle, &raft_key).unwrap().is_none() {
-            // it has been cleaned up.
-            return;
-        }
+        let raft_state: RaftLocalState =
+            match self.engine.get_msg_cf(CF_RAFT, &raft_key).unwrap() {
+                // it has been cleaned up.
+                None => return,
+                Some(value) => value,
+            };
+        let apply_key = keys::apply_state_key(region.get_id());
+        let apply_state: RaftApplyState =
+            self.engine.get_msg_cf(CF_RAFT, &apply_key).unwrap().unwrap();
```

Comment:
If restart happen when CompactLog applied (apply_state changed), but RaftlogGcTask not run.  We will get wrong first_index and dirty data after clear_meta?